File: experiments/runs/2026-09-16_16-45-11_claim-office-example-mapping_exact-sol-v1.6-test-list-dimensions-cc_opus-5-no-thinking/src/rounding.ts
/**
 * All amounts are rounded to whole G in the MHPCO's favour: an amount the
 * office receives rounds up, an amount it pays out rounds down. Intermediate
 * amounts stay fractional -- only a final premium or payout is rounded.
 */

export function roundedPremium(premium: number): number {
  return Math.ceil(premium);
}

export function roundedPayout(payout: number): number {
  return Math.floor(payout);
}

File: experiments/runs/2026-09-22_04-53-29_claim-office-python-example-mapping_exact-ptdd-v1.1-refactor-subagent-cc_opus-5-no-thinking/src/rounding.py
"""How the MHPCO rounds the money that changes hands on a policy.

The office states one rule and applies it in both directions: a final amount
is rounded to whole G in the MHPCO's own favour. Which way that falls follows
from who pays whom. Money the office receives is rounded up; money the office
hands over is rounded down. Only a final amount is rounded -- every
intermediate stays an exact fraction, so a half G is never lost early.
"""

import math


def money_received_rounded_in_mhpco_favour(amount):
    """Return an amount the MHPCO receives, rounded to whole G in its favour."""
    return math.ceil(amount)


def money_paid_out_rounded_in_mhpco_favour(amount):
    """Return an amount the MHPCO hands over, rounded to whole G in its favour."""
    return math.floor(amount)
